src/ModelContextProtocol.Core/Server/McpServerExtensions.cs

```diff
@@ -12,6 +16,18 @@ namespace ModelContextProtocol.Server;
 /// </summary>
 public static class McpServerExtensions
 {
+    /// <summary>
+    /// Caches request schemas for elicitation requests based on the type and serializer options.
+    /// </summary>
+    private static readonly ConditionalWeakTable<JsonSerializerOptions, ConcurrentDictionary<Type, ElicitRequestParams.RequestSchema>> s_elicitResultSchemaCache = new();
+
+    private static Lazy<Dictionary<string, HashSet<string>>> s_lazyElicitAllowedProperties = new(()=> new()
```

Comment:
Nit: rather than using `Lazy<T>` you can just use a nullable Dictionary field here and populate it lazily inside the method body.

Reply:
You can create a helper method if necessary to keep the main logic uncluttered.